### DISCUSSION TITLE
Hi, can you help me with this? I need to set the tooltip text for a filter icon based on the state of the filter modal. When the modal is closed, the tooltip should display "Open Filter," and when the modal is open, the tooltip should display "Close Filter."

### DISCUSSION THREAD

**Discussion by Narek1331** [filamentphp/filament#14012] [Help]
### Package

Table builder

### Package Version

v3.0.0

### How can we help you?

Hi, can you help me with this? I need to set the tooltip text for a filter icon based on the state of the filter modal. When the modal is closed, the tooltip should display "Open Filter," and when the modal is open, the tooltip should display "Close Filter."
![359465781-47b8b750-e3c8-43c0-b77f-0b506f73aae2](https://github.com/user-attachments/assets/5e357ed0-163b-4344-8804-ce323c3b7f85)

**Comment by sinnbeck** [comment]
I think you forgot to show your code? But I assume you can pass a closure ?

**Reply by Narek1331** [reply]
public static function table(Table $table): Table
    {
        return $table
            ->filtersTriggerAction(function(Action $action){
                $action->tooltip('"Open the filter" and if opened the filter chnage text to "Close the filter"')
                ->button()
                ->label('');
            })}
